integration/heapster_api_test.go

```diff
@@ -431,13 +530,16 @@ func apiTest(kubeVersion string) error {
 			continue
 		}
 		if err == nil {
+			glog.V(2).Infof("All tests passed.")
 			break
 		}
 		if attempts == 0 {
+			glog.V(2).Info("Too many attempts.")
 			return err
 		}
+		glog.V(2).Infof("Some tests failed. Retrying.")
 		attempts--
-		time.Sleep(time.Second)
+		time.Sleep(time.Second * 10)
```

Comment:
If this test doesn't complete in <10min then something is terribly broken (or your connection is dead slow). Increasing the timeout beyond 30m will not make any good.